generative/noised.wgsl

```diff
@@ -0,0 +1,86 @@
+#include "srandom.wgsl"
+
+/*
+contributors:
+    - Inigo Quilez
+    - Stevan Dedovic
+description: Returns 2D/3D value noise in the first channel and in the rest the derivatives. For more details read this nice article http://www.iquilezles.org/www/articles/gradientnoise/gradientnoise.htm
+use: noised(<vec2|vec3> space)
+options:
+    NOISED_QUINTIC_INTERPOLATION: Quintic interpolation on/off. Default is off.
+examples:
+    - /shaders/generative_noised.frag
+*/
+
+#ifndef NOISED_RANDOM2_FNC
```

Comment:
Assuming for your comment on "(manually) preprocessing" and compiling this shader I assume `#define` are still not valid WGSL. In that case let's take all this `#ifdef`, `#ifndef` and `#define` calls. Same with the mention on the commented header

Reply:
> `#define` are still not valid WGSL
That is correct.

Do you mean I should remove the `#ifdef`, `#ifndef` and `#define`? I agree, as it is not useful.

As for removing the `#include`, I would rather leave that as a comment, e.g. `// #include "../math/mod.wgsl"` so that there is a reference to the dependency chain. Or maybe a different top-level comment to simply state, "this file depends on this other file".

Thoughts?

Reply:
Exactly all preprocessing macros except '#include's